src/utils/userIP.ts

```diff
@@ -58,14 +66,29 @@ export function processUserIP(req: Request): string | null {
         return parseIP(firstEntry.for);
       } catch (e) {
         logger.warn(
-          `Request received with invalid content in "${HEADER_FORWARDED}" header.`
+          `Request received with invalid content in "${HEADER_FORWARDED}" header.`,
         );
         return null;
       }
     }
     case IPSources.XForwardedFor: {
-      logger.trace(`Sourcing User IP from "${HEADER_X_FORWARDED}" header.`);
-      return req.ip ?? null;
+      try {
+        logger.trace(`Sourcing User IP from "${HEADER_X_FORWARDED}" header.`);
+        if (isAPIGatewayProxyEvent(req)) {
+          const header = req.headers[HEADER_X_FORWARDED];
+          if (!header) return null;
+          const ip = header.split(",")[0];
+          return ip;
+        } else {
+          console.log("is  not   apigateway");
```

Comment:
Sorry if this is just for testing, but can we switch this for `logger.trace` or remove if not needed? We pay for the logs going in to splunk, and by putting it in `logger.trace` it will be silent by default unless the teams choose to turn it on for debugging.

Reply:
Sorry I forgot to remove it. It was used for my test. Removed now.